File: 240123/방향에 맞춰 이동/move-in-direction.py
x, y = 0, 0
n = int(input())

dx = [0, 1, 0, -1]
dy = [1, 0, -1, 0]
for _ in range(n):
    dir, dist = input().split()
    dist = int(dist)
    if dir == 'N':
        dir_num = 0
    elif dir == 'E':
        dir_num = 1
    elif dir == 'S':
        dir_num = 2
    else: 
        dir_num = 3

    x += dist * dx[dir_num]
    y += dist * dy[dir_num]

print(x, y)

File: 240909/방향에 맞춰 이동/move-in-direction.js
const fs = require("fs");
let input = fs.readFileSync(0).toString().trim().split("\n");

let n = Number(input[0]);
let dx = [-1, 0, 0, 1]; let dy = [0, -1, 1, 0];
let dirString = ["W", "S", "N", "E"];

let [x, y] = [0, 0];

for (let i = 1; i < input.length; i++) {
    let [dir, dist] = input[i].trim().split(" ");
    dist = Number(dist);

    dirNum = dirString.findIndex((x) => x === dir);

    x += (dx[dirNum] * dist);
    y += (dy[dirNum] * dist);
}

console.log(x, y);